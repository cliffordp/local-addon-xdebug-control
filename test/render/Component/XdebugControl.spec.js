const React = require( 'react' )
const Component = require( 'react' ).Component
const renderer = require( 'react-test-renderer' )

const context = {
	'React': React,
}

const XDebugControl = require( './../../../src/Component/XDebugControl' )( context )

describe( '<XDebugControl/>', function () {
	const defaultProps = {
		container: function () {
			return {
				readXdebugStatusAndSettings: function () {
				},
			}
		},
	}

	it( 'renders correctly provided no props', function () {
		const props = {}

		const component = renderer.create( <XDebugControl {...props}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )

	it( 'renders correctly when the site loading status is undefined', function () {
		const props = Object.assign( {site: {}}, defaultProps )

		const component = renderer.create( <XDebugControl {...props}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )

	it( 'renders correctly when the site is loading', function () {
		const props = Object.assign( {site: {loading: true}}, defaultProps )

		const component = renderer.create( <XDebugControl {...props}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )

	it( 'renders correctly when the site produces an error with no message', function () {
		const props = Object.assign( {site: {loading: false, hasError: true}}, defaultProps )

		const component = renderer.create( <XDebugControl {...props}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )

	it( 'renders correctly when the site produces an error with a message', function () {
		const props = Object.assign( {site: {loading: false, hasError: true, error: 'An error happened'}}, defaultProps )

		const component = renderer.create( <XDebugControl {...props}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )

	it( 'renders correctly when the container prop is undefined', function () {
		const props = {site: {loading: false}}

		const component = renderer.create( <XDebugControl {...props}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )

	it( 'renders correctly when environment is not custom', function () {
		const props = {
			site: {
				loading: false,
			},
			container: function () {
				return {
					readXdebugStatusAndSettings: function () {
					},
					environment: 'not-custom',
				}
			},
		}


		const component = renderer.create( <XDebugControl {...props}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )

	it( 'renders correctly when environment is custom but XDebug status is undefined', function () {
		const props = {
			site: {
				loading: false,
			},
			container: function () {
				return {
					readXdebugStatusAndSettings: function () {
					},
					environment: 'custom',
				}
			},
		}


		const component = renderer.create( <XDebugControl {...props}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )

	it( 'renders correctly when environment is custom and XDebug is active', function () {
		const props = {
			site: {
				loading: false,
			},
			container: function () {
				return {
					deactivateXdebug: function () {
					},
					activateXdebug: function () {
					},
					applyXdebugSettings: function () {
					},
					readXdebugStatusAndSettings: function () {
					},
					environment: 'custom',
				}
			},
			xdebug: {
				status: 'active',
			},
		}


		const component = renderer.create( <XDebugControl {...props}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )

	it( 'renders correctly when environment is custom and XDebug is inactive', function () {
		const props = {
			site: {
				loading: false,
			},
			container: function () {
				return {
					deactivateXdebug: function () {
					},
					activateXdebug: function () {
					},
					applyXdebugSettings: function () {
					},
					readXdebugStatusAndSettings: function () {
					},
					environment: 'custom',
				}
			},
			xdebug: {
				status: 'inactive',
			},
		}


		const component = renderer.create( <XDebugControl {...props}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )

	it( 'renders correctly when environment is custom and XDebug status is weird', function () {
		const props = {
			site: {
				loading: false,
			},
			container: function () {
				return {
					readXdebugStatusAndSettings: function () {
					},
					environment: 'custom',
				}
			},
			xdebug: {
				status: 'foo',
			},
		}


		const component = renderer.create( <XDebugControl {...props}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )
} )
