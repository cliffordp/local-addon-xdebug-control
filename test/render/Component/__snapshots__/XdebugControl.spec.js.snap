// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<XDebugControl/> renders correctly provided no props 1`] = `
<div
  className="BigLoader"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
    }
  }
>
  <div
    className="LoadingIndicator LoadingIndicator--Big"
  >
    <div
      className="LoadingIndicator_Bounce1"
    />
    <div
      className="LoadingIndicator_Bounce2"
    />
    <div
      className="LoadingIndicator_Bounce3"
    />
  </div>
</div>
`;

exports[`<XDebugControl/> renders correctly when environment is custom and XDebug is active 1`] = `
<div
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "padding": "0 5%",
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <h3
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      XDebug Controls
    </h3>
  </div>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <div
        style={
          Object {
            "marginBottom": ".5em",
            "textAlign": "center",
          }
        }
      >
        <span
          style={
            Object {
              "color": "#1FC37D",
              "fontSize": "125%",
              "textTransform": "uppercase",
            }
          }
        >
          XDeubug is active
        </span>
      </div>
      <div
        style={
          Object {
            "marginBottom": ".5em",
            "textAlign": "center",
          }
        }
      >
        <button
          className="--GrayOutline --Small Button"
          onClick={[Function]}
          style={Object {}}
        >
          Deactivate XDebug
        </button>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <div
        style={
          Object {
            "marginBottom": ".5em",
            "textAlign": "center",
          }
        }
      >
        <div
          className="FieldListWrapper"
        >
          <ul
            className="TableList Form FieldList"
            style={
              Object {
                "maxWidth": "600px",
              }
            }
          >
            <li
              className="TableListRow OptionField"
            >
              <strong>
                <span
                  className="OptionField__Title"
                >
                  remote_enable
                </span>
              </strong>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <select
                  className="OptionField__Options"
                  defaultValue="0"
                  id="remote_enable"
                  name="remote_enable"
                >
                  <option
                    value="0"
                  >
                    no
                  </option>
                  <option
                    value="1"
                  >
                    yes
                  </option>
                </select>
              </div>
            </li>
            <li
              className="TableListRow OptionField"
            >
              <strong>
                <span
                  className="OptionField__Title"
                >
                  remote_port
                </span>
              </strong>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <select
                  className="OptionField__Options"
                  defaultValue="9000"
                  id="remote_port"
                  name="remote_port"
                >
                  <option
                    value="9000"
                  >
                    9000
                  </option>
                  <option
                    value="9001"
                  >
                    9001
                  </option>
                  <option
                    value="9002"
                  >
                    9002
                  </option>
                  <option
                    value="9003"
                  >
                    9003
                  </option>
                  <option
                    value="9004"
                  >
                    9004
                  </option>
                  <option
                    value="9005"
                  >
                    9005
                  </option>
                </select>
              </div>
            </li>
            <li
              className="TableListRow OptionField"
            >
              <strong>
                <span
                  className="OptionField__Title"
                >
                  remote_autostart
                </span>
              </strong>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <select
                  className="OptionField__Options"
                  defaultValue="0"
                  id="remote_autostart"
                  name="remote_autostart"
                >
                  <option
                    value="0"
                  >
                    no
                  </option>
                  <option
                    value="1"
                  >
                    yes
                  </option>
                </select>
              </div>
            </li>
            <li
              className="TableListRow OptionField"
            >
              <strong>
                <span
                  className="OptionField__Title"
                >
                  remote_connect_back
                </span>
              </strong>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <select
                  className="OptionField__Options"
                  defaultValue="0"
                  id="remote_connect_back"
                  name="remote_connect_back"
                >
                  <option
                    value="0"
                  >
                    no
                  </option>
                  <option
                    value="1"
                  >
                    yes
                  </option>
                </select>
              </div>
            </li>
            <li
              className="TableListRow OptionField"
            >
              <strong>
                <span
                  className="OptionField__Title"
                >
                  scream
                </span>
              </strong>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <select
                  className="OptionField__Options"
                  defaultValue="0"
                  id="scream"
                  name="scream"
                >
                  <option
                    value="0"
                  >
                    no
                  </option>
                  <option
                    value="1"
                  >
                    yes
                  </option>
                </select>
              </div>
            </li>
            <li
              className="TableListRow OptionField"
            >
              <strong>
                <span
                  className="OptionField__Title"
                >
                  show_local_vars
                </span>
              </strong>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <select
                  className="OptionField__Options"
                  defaultValue="0"
                  id="show_local_vars"
                  name="show_local_vars"
                >
                  <option
                    value="0"
                  >
                    no
                  </option>
                  <option
                    value="1"
                  >
                    yes
                  </option>
                </select>
              </div>
            </li>
            <button
              className="--GrayOutline --Small Button"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                  "marginBottom": "0",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "marginTop": ".5em",
                }
              }
            >
              Apply Settings
            </button>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<XDebugControl/> renders correctly when environment is custom and XDebug is inactive 1`] = `
<div
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "padding": "0 5%",
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <h3
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      XDebug Controls
    </h3>
  </div>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <div
        style={
          Object {
            "marginBottom": ".5em",
            "textAlign": "center",
          }
        }
      >
        <span
          style={
            Object {
              "color": "#FF0000",
              "fontSize": "125%",
              "textTransform": "uppercase",
            }
          }
        >
          XDeubug is inactive
        </span>
      </div>
      <div
        style={
          Object {
            "marginBottom": ".5em",
            "textAlign": "center",
          }
        }
      >
        <button
          className="--GrayOutline --Small Button"
          onClick={[Function]}
          style={Object {}}
        >
          Activate XDebug
        </button>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <div
        style={
          Object {
            "marginBottom": ".5em",
            "textAlign": "center",
          }
        }
      >
        <div
          className="FieldListWrapper"
        >
          <ul
            className="TableList Form FieldList"
            style={
              Object {
                "maxWidth": "600px",
              }
            }
          >
            <li
              className="TableListRow OptionField"
            >
              <strong>
                <span
                  className="OptionField__Title"
                >
                  remote_enable
                </span>
              </strong>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <select
                  className="OptionField__Options"
                  defaultValue="0"
                  id="remote_enable"
                  name="remote_enable"
                >
                  <option
                    value="0"
                  >
                    no
                  </option>
                  <option
                    value="1"
                  >
                    yes
                  </option>
                </select>
              </div>
            </li>
            <li
              className="TableListRow OptionField"
            >
              <strong>
                <span
                  className="OptionField__Title"
                >
                  remote_port
                </span>
              </strong>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <select
                  className="OptionField__Options"
                  defaultValue="9000"
                  id="remote_port"
                  name="remote_port"
                >
                  <option
                    value="9000"
                  >
                    9000
                  </option>
                  <option
                    value="9001"
                  >
                    9001
                  </option>
                  <option
                    value="9002"
                  >
                    9002
                  </option>
                  <option
                    value="9003"
                  >
                    9003
                  </option>
                  <option
                    value="9004"
                  >
                    9004
                  </option>
                  <option
                    value="9005"
                  >
                    9005
                  </option>
                </select>
              </div>
            </li>
            <li
              className="TableListRow OptionField"
            >
              <strong>
                <span
                  className="OptionField__Title"
                >
                  remote_autostart
                </span>
              </strong>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <select
                  className="OptionField__Options"
                  defaultValue="0"
                  id="remote_autostart"
                  name="remote_autostart"
                >
                  <option
                    value="0"
                  >
                    no
                  </option>
                  <option
                    value="1"
                  >
                    yes
                  </option>
                </select>
              </div>
            </li>
            <li
              className="TableListRow OptionField"
            >
              <strong>
                <span
                  className="OptionField__Title"
                >
                  remote_connect_back
                </span>
              </strong>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <select
                  className="OptionField__Options"
                  defaultValue="0"
                  id="remote_connect_back"
                  name="remote_connect_back"
                >
                  <option
                    value="0"
                  >
                    no
                  </option>
                  <option
                    value="1"
                  >
                    yes
                  </option>
                </select>
              </div>
            </li>
            <li
              className="TableListRow OptionField"
            >
              <strong>
                <span
                  className="OptionField__Title"
                >
                  scream
                </span>
              </strong>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <select
                  className="OptionField__Options"
                  defaultValue="0"
                  id="scream"
                  name="scream"
                >
                  <option
                    value="0"
                  >
                    no
                  </option>
                  <option
                    value="1"
                  >
                    yes
                  </option>
                </select>
              </div>
            </li>
            <li
              className="TableListRow OptionField"
            >
              <strong>
                <span
                  className="OptionField__Title"
                >
                  show_local_vars
                </span>
              </strong>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <select
                  className="OptionField__Options"
                  defaultValue="0"
                  id="show_local_vars"
                  name="show_local_vars"
                >
                  <option
                    value="0"
                  >
                    no
                  </option>
                  <option
                    value="1"
                  >
                    yes
                  </option>
                </select>
              </div>
            </li>
            <button
              className="--GrayOutline --Small Button"
              onClick={[Function]}
              style={
                Object {
                  "display": "block",
                  "marginBottom": "0",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "marginTop": ".5em",
                }
              }
            >
              Apply Settings
            </button>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<XDebugControl/> renders correctly when environment is custom and XDebug status is weird 1`] = `
<div
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "padding": "0 5%",
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <h3
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      XDebug Controls
    </h3>
  </div>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <span
      style={
        Object {
          "color": "#FF0000",
          "fontSize": "125%",
          "textTransform": "uppercase",
        }
      }
    >
      XDebug status is undefined
    </span>
  </div>
</div>
`;

exports[`<XDebugControl/> renders correctly when environment is custom but XDebug status is undefined 1`] = `
<div
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "padding": "0 5%",
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <h3
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      XDebug Controls
    </h3>
  </div>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <span
      style={
        Object {
          "color": "#FF0000",
          "fontSize": "125%",
          "textTransform": "uppercase",
        }
      }
    >
      XDebug status is undefined
    </span>
  </div>
</div>
`;

exports[`<XDebugControl/> renders correctly when environment is not custom 1`] = `
<div
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "padding": "0 5%",
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": "1em",
        "marginTop": "1em",
      }
    }
  >
    <h3
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Only available in custom installations!
    </h3>
  </div>
</div>
`;

exports[`<XDebugControl/> renders correctly when the container prop is undefined 1`] = `
<div
  className="BigLoader"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
    }
  }
>
  <div
    className="LoadingIndicator LoadingIndicator--Big"
  >
    <div
      className="LoadingIndicator_Bounce1"
    />
    <div
      className="LoadingIndicator_Bounce2"
    />
    <div
      className="LoadingIndicator_Bounce3"
    />
  </div>
</div>
`;

exports[`<XDebugControl/> renders correctly when the site is loading 1`] = `
<div
  className="BigLoader"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
    }
  }
>
  <div
    className="LoadingIndicator LoadingIndicator--Big"
  >
    <div
      className="LoadingIndicator_Bounce1"
    />
    <div
      className="LoadingIndicator_Bounce2"
    />
    <div
      className="LoadingIndicator_Bounce3"
    />
  </div>
</div>
`;

exports[`<XDebugControl/> renders correctly when the site loading status is undefined 1`] = `
<div
  className="BigLoader"
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
    }
  }
>
  <div
    className="LoadingIndicator LoadingIndicator--Big"
  >
    <div
      className="LoadingIndicator_Bounce1"
    />
    <div
      className="LoadingIndicator_Bounce2"
    />
    <div
      className="LoadingIndicator_Bounce3"
    />
  </div>
</div>
`;

exports[`<XDebugControl/> renders correctly when the site produces an error with a message 1`] = `
<section
  className="Error"
  style={
    Object {
      "border": "2px solid #FF0000",
      "color": "#FF0000",
      "padding": ".25em",
    }
  }
>
  <div
    className="Error__Message"
  >
    <div
      style={
        Object {
          "fontSize": "120%",
          "marginBottom": "1em",
          "textAlign": "center",
        }
      }
    >
      Error generated in 
      <span
        className="Error__Message__Source"
      >
        Site Container
      </span>
    </div>
    <div
      className="Error__Message__Text"
    >
      An error happened
    </div>
    <div
      className="Error__Message__CTA"
      style={
        Object {
          "fontSize": "120%",
          "marginTop": "1em",
          "textAlign": "center",
        }
      }
    >
      <section>
        <p>
          Try to quit and restart Local. Yes: turning it off and on again 
          <em>
            might
          </em>
           help.
        </p>
        <p>
          If the issue persists please open an issue 
          <a
            href="https://github.com/lucatume/local-addon-xdebug-control/issues/new"
          >
            on the addon repository
          </a>
           containing the message above, how you got here, what you were trying to do and your setup.
        </p>
      </section>
    </div>
  </div>
</section>
`;

exports[`<XDebugControl/> renders correctly when the site produces an error with no message 1`] = `
<section
  className="Error"
  style={
    Object {
      "border": "2px solid #FF0000",
      "color": "#FF0000",
      "padding": ".25em",
    }
  }
>
  <div
    className="Error__Message"
  >
    <div
      style={
        Object {
          "fontSize": "120%",
          "marginBottom": "1em",
          "textAlign": "center",
        }
      }
    >
      Error generated in 
      <span
        className="Error__Message__Source"
      >
        Site Container
      </span>
    </div>
    <div
      className="Error__Message__Text"
    >
      no message provided
    </div>
    <div
      className="Error__Message__CTA"
      style={
        Object {
          "fontSize": "120%",
          "marginTop": "1em",
          "textAlign": "center",
        }
      }
    >
      <section>
        <p>
          Try to quit and restart Local. Yes: turning it off and on again 
          <em>
            might
          </em>
           help.
        </p>
        <p>
          If the issue persists please open an issue 
          <a
            href="https://github.com/lucatume/local-addon-xdebug-control/issues/new"
          >
            on the addon repository
          </a>
           containing the message above, how you got here, what you were trying to do and your setup.
        </p>
      </section>
    </div>
  </div>
</section>
`;
